Source/WebCore/xml/XPathValue.cpp

```diff
@@ -117,7 +117,7 @@ String Value::toString() const
     case Type::NodeSet:
         if (m_data->nodeSet.isEmpty())
             return emptyString();
-        return stringValue(m_data->nodeSet.firstNode());
+        return stringValue(RefPtr { m_data->nodeSet.firstNode() }.get());
```

Comment:
Would be nice to introduce a `protectedFirstNode()` for this one.

Reply:
As far as I can tell this is the only call to `firstNode()` that needs protecting and it seems unlikely we'll introduce new XPath code anytime soon. Happy to do it, but it seemed superfluous to me.